src/transformers/models/conditional_detr/image_processing_conditional_detr.py

```diff
@@ -1356,7 +1358,8 @@ def post_process_object_detection(
                 )
 
         prob = out_logits.sigmoid()
-        topk_values, topk_indexes = torch.topk(prob.view(out_logits.shape[0], -1), 100, dim=1)
+        k_value = min(top_k, prob.view(out_logits.shape[0], -1).size(1))
+        topk_values, topk_indexes = torch.topk(prob.view(out_logits.shape[0], -1), k_value, dim=1)
```

Comment:
I would reshape `prob` here to make the code a bit clearer and reduce the repeated view operation. 

```suggestion
        prob = prob.view(out_logits.shape[0], -1)
        k_value = min(top_k, prob.size(1))
        topk_values, topk_indexes = torch.topk(prob, k_value, dim=1)
```